src/SoftwareSerial.h

```diff
@@ -32,11 +32,76 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 // If only one tx or rx wanted then use this as parameter for the unused pin
 constexpr int SW_SERIAL_UNUSED_PIN = -1;
 
-enum SoftwareSerialConfig {
-	SWSERIAL_5N1 = 0,
-	SWSERIAL_6N1,
-	SWSERIAL_7N1,
-	SWSERIAL_8N1,
+// Options for SoftwareSerialConfig
+constexpr uint8_t SWSERIAL_NB_BIT_MASK       = 0B00011000;
+constexpr uint8_t SWSERIAL_NB_BIT_5          = 0B00000000;
+constexpr uint8_t SWSERIAL_NB_BIT_6          = 0B00001000;
+constexpr uint8_t SWSERIAL_NB_BIT_7          = 0B00010000;
+constexpr uint8_t SWSERIAL_NB_BIT_8          = 0B00011000;
+
+constexpr uint8_t SWSERIAL_PARITY_CONF_MASK  = 0B00000111;
+constexpr uint8_t SWSERIAL_PARITY_CONF_NONE  = 0B00000000;
+constexpr uint8_t SWSERIAL_PARITY_CONF_EVEN  = 0B00000010;
+constexpr uint8_t SWSERIAL_PARITY_CONF_ODD   = 0B00000011;
+constexpr uint8_t SWSERIAL_PARITY_CONF_MARK  = 0B00000100;
+constexpr uint8_t SWSERIAL_PARITY_CONF_SPACE = 0B00000101;
+
+constexpr uint8_t SWSERIAL_NB_STOP_BIT_MASK  = 0B01100000;
+constexpr uint8_t SWSERIAL_NB_STOP_BIT_0     = 0B00000000;
```

Comment:
same here, SWSERIAL_0_STOP_BITS, SWSERIAL_STOP_BITS_0 feels easier to digest for me